MachineCodingRound/main.cpp

```diff
@@ -0,0 +1,136 @@
+#include <iostream>
+#include<bits/stdc++.h>
+
+using namespace std;
+
+
+class snake
+{
+public:
+    int head;
+    int tail;
+
+    snake(int headOfSnakes, int tailOfSnakes)
+    {
+        head = headOfSnakes;
+        tail = tailOfSnakes;
+    }
+
+};
+
+class ladder
+{
+public:
+    int start_positions;
+    int end_positions;
+    ladder(int start,int end)
+    {
+        start_positions = start;
+        end_positions = end;
+    }
+};
+
+class player
+{
+public:
+    string playerName;
+    int curPositions;
+    player(string name)
+    {
+        playerName = name;
+        curPositions =0;
+    }
+};
+
+class Dice {
+    public:
+        Dice(){
+             srand(time(NULL));
+        }
+    int roolDice(){
+
+        return rand()%6+1;
+    }
+};
+class snakeAndLadderGame
+{   public:
+    int numberOfSnakes;
```

Comment:
Why have numberOf... as class property. You anyway have the vectors to get the size when required.